src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+```
+
+It is worth mentioning that, since we have being working on this dataset over the latest several weeks, this project skips the Exploratory Data Analysis (EDA) phase as it is already available in previous reports.
+
+## Milestone 1: Exploration Phase
+### 1. Building the dataset
+
+#### 1.1 Filtering the data to orders with at least 5 items
+
+Firstly, we filter the data to only those orders with at least 5 items to build a dataset to work with, since they are our target. 
+
+
+```python
+# Group data by order_id and sum the outcome (A product is in the order if its outcome is 1)
+order_size = data.groupby('order_id').outcome.sum()
+
+# Identify orders with size at least 5 and extract the order_id of these big orders
+big_orders = order_size[order_size >= 5].index 
+
+# Filter data to only include rows where order_id is in the list of big_orders
+filtered_data = data.loc[lambda x: x.order_id.isin(big_orders)]
+```
+
+
+```python
+filtered_data.shape
+```
+
+
+
+
+    (2163953, 27)
+
+
+
+#### 1.2 Splitting the data
+
+
+```python
+# Get how many orders are performed each day
+daily_orders = filtered_data.groupby('order_date').order_id.nunique()
+daily_orders.head()
+```
+
+
+
+
+    order_date
+    2020-10-05     3
+    2020-10-06     7
+    2020-10-07     6
+    2020-10-08    12
+    2020-10-09     4
+    Name: order_id, dtype: int64
+
+
+
+
+```python
+plt.plot(daily_orders, label="Daily orders")
+plt.title("Daily Orders")
+```
+
+
+
+
+    Text(0.5, 1.0, 'Daily Orders')
+
+
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_21_1.png)
+    
+
+
+**Important**: It's evident that there is a significant time-based aspect, signaling changes in patterns and business evolution over time. Consequently, it is prudent to implement a temporal split. This approach not only captures the dynamic nature of the data but also ensures that the same order does not appear in both the training and testing sets, thereby preventing information leakage.
+
+
+```python
+cum_sum_daily_orders = daily_orders.cumsum() / daily_orders.sum()
+
+train_val_cut = cum_sum_daily_orders[cum_sum_daily_orders <= 0.7].idxmax()
+val_test_cut = cum_sum_daily_orders[cum_sum_daily_orders <= 0.9].idxmax()
+
+print("Train set from: ", cum_sum_daily_orders.index.min())
+print("Train set until: ", train_val_cut)
+print("Validation set from: ", val_test_cut)
+print("Test set until: ", cum_sum_daily_orders.index.max())
+```
+
+    Train set from:  2020-10-05 00:00:00
+    Train set until:  2021-02-04 00:00:00
+    Validation set from:  2021-02-22 00:00:00
+    Test set until:  2021-03-03 00:00:00
+
+
+It is important to consider what is the minimum business cicle, that is, the minimum temporal window that allows us to capture the complete dynamics of the data we want to model. Even though the validation and test splits don't comprise many days, we consider these should be sufficient for our case.
```

Comment:
In real life we should have an understanding of the business or talk to business people to do this assessment. For this project, just noticing this is perfect!